components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -896,6 +896,13 @@ private static User getUserFromTenant(String username, String userId, int tenant
                 user = new User(userStoreManager.getUser(null, username));
             } else if (userId != null && userStoreManager.isExistingUserWithID(userId)) {
                 user = new User(userStoreManager.getUser(userId, null));
+            } else if (username != null && userStoreManager.isExistingUserWithID(username)) {
```

Comment:
since a username != null check has been done above, we can refactor the code to avoid checking the same condition again.